HW2/task.js

```diff
@@ -5,32 +5,142 @@
 // 1. Show the practical difference in working with variables 
 // that are defined by var, let or const. Try to describe all possible situations.
 
+// 1.1 We can redefine variables created only by var
 
+var vVariable = 11;
+var vVariable = 22;
+console.log('everything works correctly');
 
-// 2. Write a good example of how Hoisting works
-
+let lVariable = 33;
+let lVariable = 44; // Error: Uncaught SyntaxError: Identifier 'lVariable' has already been declared
+lVariable = 444; // but can be overwritteng
+console.log('code execution will not reach this point, there will be an error earlier');
 
+const cVariable = 55;
+const cVariable = 66; // Error: Uncaught SyntaxError: Identifier 'cVariable' has already been declared!
+cVariable = 666; // Error can not be overwritten!
+console.log('same error as when declared by let');
+
+// 1.2 ......
+//also we can see zone of visibility variable created by var-let-const
+function giveValue(valueVar) {
+    if (valueVar) {
+        let a = 1
+        var b = 2
+        const c = 3
+    }
+    return b //undefined - because var variable is global, and appopriation is only will be inside
+    //return a // error 
+    //return c // also error because let and const have block visibility (can't be initialized above, for exampe like(var)
+}
+console.log(giveValue(false))
 
+// 2. Write a good example of how Hoisting works
+console.log(reVar)
+reVar = 'Oh!'
+console.log(reVar)
+console.log(nameGr(reVar, 'Mark'))
```

Comment:
don't forget to put a semicolon!

don't forget to write what exactly the console will display the log.
 example
 
 console.log(reVar) // 'Oh!'

Reply:
O! i understood